tests/ui/borrowck/regions-bound-missing-bound-in-impl.rs

```diff
@@ -17,11 +17,11 @@ pub trait Foo<'a, 't> {
 
 impl<'a, 't> Foo<'a, 't> for &'a isize {
     fn no_bound<'b:'a>(self, b: Inv<'b>) {
-        //~^ ERROR lifetime parameters or bounds on method `no_bound` do not match
+        //~^ ERROR lifetime parameters do not match the trait definition, since they differ in late-boundedness
```

Comment:
Hmm, brief thought as I skim this, but how often do we actually use the terminology "late-bound"/"early-bound" in error mesasges? In general, "late-boundedness" is kind of weird.

Reply:
no idea, but also we do need a precise term here. hiding this detail feels like we're obscuring the whole fact that this error arises.

Reply:
AFAIK (I've once grepped for this) in diagnostics we only ever mention boundedness in the case where a users tries to bind a late-bound var early:

```rs
fn main() { fn f<'a>() {} f::<'static>; }
//~^ ERROR cannot specify lifetime arguments explicitly if late bound lifetime parameters are present [E0794]
```

Reply:
The reference also uses the terms without any definition 💀 https://github.com/rust-lang/reference/blob/48ad733ce25b71385e917cab0383ed3407603490/src/types/function-item.md?plain=1#L10

Reply:
I kind of think that maybe just exclude the "since they differ in late-boundedness" part in the main error text.

We could also just kind of "lie" a bit and state that the reason there is an error is because of the extra bounds (which is true, but not precise).

Reply:
Yeah, having "late-bound" and "early-bound" in the note is fine because of googleability, but something about the "lateboundedness" term rubs me the wrong way - this is mostly just bikeshedding, because I don't immediately have a better answer.

Reply:
Reworded this to just say "lifetime parameters do not match the trait definition", and added a note saying "lifetime parameters differ in whether they are early- or late-bound".